src/Modules/Workspaces/Commands.Workspaces/GetPowerBIWorkspace.cs

```diff
@@ -47,20 +49,29 @@ public class GetPowerBIWorkspace : PowerBIClientCmdlet, IUserScope, IUserFilter,
         public PowerBIUserScope Scope { get; set; } = PowerBIUserScope.Individual;
 
         [Parameter(Mandatory = false, ParameterSetName = ListParameterSetName)]
+        [Parameter(Mandatory = false, ParameterSetName = AllParameterSetName)]
         public string Filter { get; set; }
 
         [Parameter(Mandatory = false, ParameterSetName = ListParameterSetName)]
+        [Parameter(Mandatory = false, ParameterSetName = AllParameterSetName)]
         public string User { get; set; }
 
         [Parameter(Mandatory = false, ParameterSetName = ListParameterSetName)]
+        [Parameter(Mandatory = false, ParameterSetName = AllParameterSetName)]
         public SwitchParameter Deleted { get; set; }
 
         [Parameter(Mandatory = false, ParameterSetName = ListParameterSetName)]
+        [Parameter(Mandatory = false, ParameterSetName = AllParameterSetName)]
         public SwitchParameter Orphaned { get; set; }
 
+        [Parameter(Mandatory = true, ParameterSetName = AllParameterSetName)]
+        [Parameter(Mandatory = false, ParameterSetName = IdParameterSetName)]
+        [Parameter(Mandatory = false, ParameterSetName = NameParameterSetName)]
+        public SwitchParameter All { get; set; }
```

Comment:
Hard to see in the review, why do we need a new parameter set just for All instead of it being on List?

Reply:
Also discussed offline. This was needed because we want to have some restrictions enforced on what other parameters we want to use along with -All. For example -First and -Skip cannot be used with -All. -First, -Skip are part of the ListParameterSet. So if we have -All as a ListParameterSet, then -First and -Skip will be simply ignored without letting the consumer know that he is not supposed to provide -First and -Skip along with -All. With the new 'AllParameterSet'  the validation is taken care by that implicitly and whenever consumer provides -First and -Skip along with -All it shows an error to him and stops the transaction.